### DISCUSSION TITLE
ERROR: Service 'react-storefront' failed to build : Build failed

### DISCUSSION THREAD

**Discussion by giannineves** [saleor/saleor#9736] [Deployment] [answered]
Can anybody help me?

#12 28.91 > react-storefront@0.1.0 prepare /app
#12 28.91 > husky install
#12 28.91
#12 28.96 fatal: not a git repository (or any of the parent directories): .git
#12 28.97
#12 28.97 dependencies:
#12 28.97 + @apollo/client 3.5.10
#12 28.97 + @formatjs/swc-plugin 1.4.0
#12 28.97 + @formatjs/ts-transformer 3.9.2
#12 28.97 + @graphql-codegen/typescript-apollo-client-helpers 2.1.15
#12 28.97 + @headlessui/react 1.5.0
#12 28.97 + @heroicons/react 1.0.5
#12 28.97 + @saleor/sdk 0.4.4
#12 28.97 + @stripe/react-stripe-js 1.7.0
#12 28.97 + @stripe/stripe-js 1.26.0
#12 28.97 + @tailwindcss/aspect-ratio 0.4.0
#12 28.97 + @tailwindcss/forms 0.5.0
#12 28.97 + @tailwindcss/typography 0.5.2
#12 28.97 + clsx 1.1.1
#12 28.97 + dotenv 16.0.0
#12 28.97 + dotenv-cli 5.1.0
#12 28.97 + editorjs-blocks-react-renderer 1.2.4
#12 28.97 + graphql 16.3.0
#12 28.97 + next 12.1.4
#12 28.97 + next-seo 5.3.0
#12 28.97 + next-sitemap 2.4.3
#12 28.97 + next-usequerystate 1.7.0
#12 28.97 + react 18.0.0
#12 28.97 + react-dom 18.0.0
#12 28.97 + react-intl 5.24.6
#12 28.97 + tailwind-scrollbar-hide 1.1.7
#12 28.97
#12 28.97 devDependencies:
#12 28.97 + @formatjs/cli 4.8.2
#12 28.97 + @graphql-codegen/add 3.1.1
#12 28.97 + @graphql-codegen/cli 2.6.2
#12 28.97 + @graphql-codegen/introspection 2.1.1
#12 28.97 + @graphql-codegen/typescript 2.4.8
#12 28.97 + @graphql-codegen/typescript-operations 2.3.5
#12 28.97 + @graphql-codegen/typescript-react-apollo 3.2.11
#12 28.97 + @next/bundle-analyzer 12.1.4
#12 28.97 + @types/qs 6.9.7
#12 28.97 + @types/react 17.0.43
#12 28.97 + @typescript-eslint/eslint-plugin 5.17.0
#12 28.97 + @typescript-eslint/parser 5.17.0
#12 28.97 + autoprefixer 10.4.2
#12 28.97 + eslint 8.12.0
#12 28.97 + eslint-config-airbnb 19.0.4
#12 28.97 + eslint-config-next 12.1.0
#12 28.97 + eslint-config-prettier 8.4.0
#12 28.97 + eslint-plugin-formatjs 3.0.0
#12 28.97 + eslint-plugin-import 2.25.4
#12 28.97 + eslint-plugin-jsx-a11y 6.5.1
#12 28.97 + eslint-plugin-react 7.29.4
#12 28.97 + eslint-plugin-react-hooks 4.3.0
#12 28.97 + eslint-plugin-simple-import-sort 7.0.0
#12 28.97 + husky 7.0.4
#12 28.97 + pathpida 0.18.1
#12 28.97 + postcss 8.4.12
#12 28.97 + prettier 2.6.2
#12 28.97 + react-hook-form 7.29.0
#12 28.97 + react-use 17.3.2
#12 28.97 + tailwindcss 3.0.23
#12 28.97 + typescript 4.6.3
#12 28.97
#12 28.97  ERR_PNPM_PEER_DEP_ISSUES  Unmet peer dependencies
#12 28.97
#12 28.97 .
#12 28.97 ├─┬ @graphql-codegen/typescript-react-apollo
#12 28.97 │ └── ✕ missing peer graphql-tag@^2.0.0
#12 28.97 ├─┬ @graphql-codegen/typescript
#12 28.97 │ └─┬ @graphql-codegen/visitor-plugin-common
#12 28.97 │   └─┬ @graphql-tools/relay-operation-optimizer
#12 28.97 │     └─┬ relay-compiler
#12 28.97 │       └── ✕ unmet peer graphql@^15.0.0: found 16.3.0
#12 28.97 ├─┬ @saleor/sdk
#12 28.97 │ ├── ✕ unmet peer graphql@^15.5.0: found 16.3.0
#12 28.97 │ └── ✕ unmet peer react@"^16.8.0 || ^17.0.0": found 18.0.0
#12 28.97 ├─┬ react-use
#12 28.97 │ ├── ✕ unmet peer react@"^16.8.0  || ^17.0.0": found 18.0.0
#12 28.97 │ └── ✕ unmet peer react-dom@"^16.8.0  || ^17.0.0": found 18.0.0
#12 28.97 ├─┬ @apollo/client
#12 28.97 │ └── ✕ unmet peer react@"^16.8.0 || ^17.0.0": found 18.0.0
#12 28.97 ├─┬ @stripe/react-stripe-js
#12 28.97 │ ├── ✕ unmet peer react@"^16.8.0 || ^17.0.0": found 18.0.0
#12 28.97 │ └── ✕ unmet peer react-dom@"^16.8.0 || ^17.0.0": found 18.0.0
#12 28.97 ├─┬ editorjs-blocks-react-renderer
#12 28.97 │ └─┬ html-react-parser
#12 28.97 │   └── ✕ unmet peer react@"0.14 || 15 || 16 || 17": found 18.0.0
#12 28.97 └─┬ react-intl
#12 28.97   └── ✕ unmet peer react@"^16.3.0 || 17": found 18.0.0
#12 28.97 Peer dependencies that should be installed:
#12 28.97   graphql-tag@^2.0.0
#12 28.97
------
executor failed running [/bin/sh -c pnpm install]: exit code: 1
ERROR: Service 'react-storefront' failed to build : Build failed

**Comment by timuric** [comment]
@giannineves does it work if you try `npx saleor storefront create --demo` ?

**Reply by giannineves** [reply]
Hi @timuric,

I did not use npx. I am trying to run the docker installation. I've tried Ubuntu, Alpine, and now Windows. All with the same error. Where I am suppose to run the NPX command?

Thanks,

**Reply by timuric** [reply]
@giannineves this command would bootstrap a storefront using a free Saleor cloud backend. You can install npx with npm `npm install -g npx`

**Comment by giannineves** [comment, marked as answer]
I corrected all dependencies manually in package.json.
and installed graphql-tag manually in the docker file.

**Comment by dhenryFabrice** [comment]
Hi,

Would it be possible to get the corrected files? 

Thanks

**Comment by kimus** [comment]
I've done this:

```diff
diff --git a/package.json b/package.json
index d7233d7..4071f12 100644
--- a/package.json
+++ b/package.json
@@ -35,13 +35,13 @@
     "dotenv": "^16.0.0",
     "dotenv-cli": "^5.1.0",
     "editorjs-blocks-react-renderer": "^1.2.4",
-    "graphql": "16.3.0",
+    "graphql": "15.7.2",
     "next": "12.1.4",
     "next-seo": "^5.3.0",
     "next-sitemap": "^2.4.3",
     "next-usequerystate": "1.7.0",
-    "react": "18.0.0",
-    "react-dom": "18.0.0",
+    "react": "17.0.2",
+    "react-dom": "17.0.2",
     "react-hook-form": "^7.29.0",
     "react-intl": "^5.24.6",
     "react-use": "^17.3.2",
@@ -55,6 +55,7 @@
     "@graphql-codegen/typescript": "2.4.8",
     "@graphql-codegen/typescript-operations": "2.3.5",
     "@graphql-codegen/typescript-react-apollo": "3.2.11",
+    "graphql-tag": "^2.0.0",
     "@next/bundle-analyzer": "^12.1.4",
     "@types/qs": "^6.9.7",
     "@types/react": "17.0.43",
```
